### PR TITLE
DA-535: RAG with Couchbase and Claude(by Anthropic) Notebook

### PR DESCRIPTION
**Pull request by teetangh**

### CODE REVIEW

**Comment by nithishr** [comment]
@teetangh Once you verify that the comments in other PRs are addressed in this one, please ask for review again.

**Review by nithishr — CHANGES_REQUESTED** [review]

File: claudeai/RAG_with_Couchbase_and_Claude.ipynb [hunk 1]
```diff
@@ -0,0 +1,3528 @@
...
+      ],
+      "metadata": {
+        "id": "kNdImxzypDlm"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Setting the Stage: Installing Necessary Libraries\n",
+        "To build our semantic search engine, we need a robust set of tools. The libraries we install handle everything from connecting to databases to performing complex machine learning tasks. Each library has a specific role: Couchbase libraries manage database operations, LangChain handles AI model integrations, and Claude provides advanced AI models for generating embeddings and understanding natural language. By setting up these libraries, we ensure our environment is equipped to handle the data-intensive and computationally complex tasks required for semantic search."
```
[the remark this answers is not in the document]
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude.ipynb:2088]: Changed to OpenAI embeddings everywhere

File: claudeai/RAG_with_Couchbase_and_Claude.ipynb [hunk 2]
```diff
@@ -0,0 +1,3528 @@
...
+        }
+      }
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Introduction\n",
+        "In this guide, we will walk you through building a powerful semantic search engine using Couchbase as the backend database, [OpenAi](https://openai.com/) as the AI-powered embedding and [Claude](https://claude.ai/) as the language model provider. Semantic search goes beyond simple keyword matching by understanding the context and meaning behind the words in a query, making it an essential tool for applications that require intelligent information retrieval. This tutorial is designed to be beginner-friendly, with clear, step-by-step instructions that will equip you with the knowledge to create a fully functional semantic search system from scratch."
```
[the remark this answers is not in the document]
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude.ipynb:2078]: changed

File: claudeai/RAG_with_Couchbase_and_Claude.ipynb [hunk 3]
```diff
@@ -0,0 +1,3528 @@
...
+            "2024-08-29 13:35:10,138 - INFO - Successfully loaded TREC dataset with 1000 samples\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Creating Claude Embeddings\n",
+        "Embeddings are at the heart of semantic search. They are numerical representations of text that capture the semantic meaning of the words and phrases. Unlike traditional keyword-based search, which looks for exact matches, embeddings allow our search engine to understand the context and nuances of language, enabling it to retrieve documents that are semantically similar to the query, even if they don't contain the exact keywords. By creating embeddings using Claude, we equip our search engine with the ability to understand and process natural language in a way that's much closer to how humans understand language. This step transforms our raw text data into a format that the search engine can use to find and rank relevant documents.\n",
```
[the remark this answers is not in the document]
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude.ipynb:3048]: Changed to OpenAI embeddings everywhere

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb [hunk 4]
```diff
@@ -0,0 +1,3528 @@
...
+        "id": "Yv8kWcuf8mLx"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Loading Sensitive Informnation\n",
+        "In this section, we prompt the user to input essential configuration settings needed for integrating Couchbase with Cohere's API. These settings include sensitive information like API keys, database credentials, and specific configuration names. Instead of hardcoding these details into the script, we request the user to provide them at runtime, ensuring flexibility and security.\n",
```
Comment (nithishr) [inline claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb:2319]: Cohere?

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb [hunk 5]
```diff
@@ -0,0 +1,3528 @@
...
+          "text": [
+            "2024-08-29 13:35:27,788 - INFO - Successfully created cache\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Creating the OpenAi Language Model (LLM)\n",
```
Comment (nithishr) [inline claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb:3219]: OpenAI

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb [hunk 6]
```diff
@@ -0,0 +1,3528 @@
...
+        "Language models are AI systems that are trained to understand and generate human language. We'll be using OpenAI's language model to process user queries and generate meaningful responses. This model is a key component of our semantic search engine, allowing it to go beyond simple keyword matching and truly understand the intent behind a query. By creating this language model, we equip our search engine with the ability to interpret complex queries, understand the nuances of language, and provide more accurate and contextually relevant responses.\n",
```
Comment (nithishr) [inline claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb:3220]: Is it using OpenAI's LLM?

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb [hunk 7]
```diff
@@ -0,0 +1,3528 @@
...
+            "Distance: 0.7857, Text: When did World War I start ?\n",
+            "Distance: 0.7842, Text: What caused Harry Houdini 's death ?\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Retrieval-Augmented Generation (RAG) with Couchbase and Langchain\n",
```
Comment (nithishr) [inline claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb:3332]: LangChain

**Review by nithishr — CHANGES_REQUESTED** [review]

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb (hunk 5 above)
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb]: Fixed and Changed to Claude 3.5 Sonnet Language model.

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb (hunk 6 above)
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb]: Fixed and Changed to Claude 3.5 Sonnet Language model.

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb (hunk 7 above)
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb]: Fixed

File: claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb (hunk 4 above)
Comment (teetangh) [reply claudeai/RAG_with_Couchbase_and_Claude(by_Anthropic).ipynb]: Removed

**Review by nithishr — APPROVED** [review]

**Pull request merged**